File: BOJ(Java 공부용)/10814.java
import java.io.*;
import java.util.ArrayList;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().split(" ")[0]);
        ArrayList<p> v = new ArrayList<>();
        for(int i=0;i<n;i++) {
            String[] input = br.readLine().split(" ");
            int age = Integer.parseInt(input[0]);
            v.add(new p(age,i,input[1]));
        }
        v.sort((a, b) -> {
            if (a.age < b.age) return -1;
            else if (a.age > b.age) return 1;
            else {
                if(a.index<b.index) return -1;
                else if(a.index>b.index) return 1;
                else return 0;
            }
        });
        for(int i=0;i<v.size();i++) {
            bw.write(v.get(i).age + " "+ v.get(i).name+"\n");
        }
        bw.flush();
        bw.close();
    }
    private static class p{
        int age;
        int index;
        String name;
        private p(int age,int index,String name) {
            this.name = name;
            this.index = index;
            this.age = age;
        }
    }
}

File: BOJ(Kotlin)/10814.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import kotlin.math.max
import kotlin.math.min

var ans = 0
fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    var list : MutableList<User> = mutableListOf()
    for (i in 1..n) {
        val inputList : List<String> = br.readLine().split(" ")
        list.add(User(inputList.get(0).toInt(),i,inputList.get(1)))
    }
   list.sortWith(Comparator<User> {o1, o2 ->
       when {
           o1.age < o2.age -> -1
           o1.age == o2.age -> {
               when {
                   o1.index < o2.index -> -1
                   o1.index  == o2.index -> 0
                   else -> 1
               }
           }
           else -> 1
       }
   })
    for (value in list) {
        bw.write(value.age.toString() + " "+ value.name+"\n")
    }
    bw.flush()
    bw.close()
}

class User(val age : Int, val index : Int, val name : String) {

}